src/components/VersionRestartModal/VersionRestartModal.tsx

```diff
@@ -190,7 +263,29 @@ const selectedArray = (selected: selectedStrings) => {
   return out;
 };
 
-const HR = styled("hr")`
+const selectTasksTotal = (selectedTasks: patchSelectedTasks) => {
+  let total = 0;
+  Object.keys(selectedTasks).forEach((selected) => {
+    total += selectedArray(selectedTasks[selected]).length;
+  });
+  return total;
+};
+
+const getTaskIds = (selectedTasks: patchSelectedTasks) => {
+  const taskIds: VersionToRestart[] = [];
+  Object.keys(selectedTasks).forEach((selected) => {
+    const tasksArray = selectedArray(selectedTasks[selected]);
+
+    const versionToRestart: VersionToRestart = {
+      versionId: selected,
+      taskIds: tasksArray,
+    };
+    taskIds.push(versionToRestart);
+  });
+  return taskIds;
+};
```

Comment:
Something like this.
```suggestion
const getTaskIds = (selectedTasks: patchSelectedTasks) => (
Object.entries(selectedTasks).map((selectedTask) => ( {
   versionId: selectedTask[0],
   taskIds: selectedArray(selectedTask[1])
})
);
```